include/boost/math/interpolators/detail/bezier_polynomial_detail.hpp

```diff
@@ -12,6 +12,15 @@
 
 namespace boost::math::interpolators::detail {
 
+namespace {
+    template <class RandomAccessContainer>
+    static RandomAccessContainer& get_storage()
+    {
+        static thread_local RandomAccessContainer the_storage;
```

Comment:
This may run into issues with Mingw since it only supports using `thread_local` with trivial types. I think the following should work:

```
static
#ifndef BOOST_MATH_NO_THREAD_LOCAL_WITH_NON_TRIVIAL_TYPES
    BOOST_MATH_THREAD_LOCAL
#endif
RandomAccessContainer the_storage;
```

@jzmaddock feel free to correct me if I am wrong.

Reply:
@mborland : I've restricted to C++-17; will that not help?

Reply:
It probably will; I forget which versions of GCC it impacted but they are older. If the CI is clean I wouldn't worry about it.

Reply:
I'm afraid I've had issues up to and including gcc-10 with this.  Sometimes you hit it and sometimes you don't :(

There are also those platforms which have zero thread support (no thread_local, <mutex> etc).

Reply:
For example: https://github.com/boostorg/math/issues/621